Cython/Compiler/PyrexTypes.py

```diff
@@ -4599,12 +4599,24 @@ def best_match(arg_types, functions, pos=None, env=None, args=None):
             errors.append((func, error_mesg))
             continue
         if func_type.templates:
+            # For any argument/parameter pair A/P, if P is a forwarding reference,
+            # use lvalue-reference-to-A for deduction in place of A when the
+            # function call argument is an lvalue. See:
+            # https://en.cppreference.com/w/cpp/language/template_argument_deduction#Deduction_from_a_function_call
+            arg_types_for_deduction = copy.copy(arg_types)
```

Comment:
Let's keep this simple.
```suggestion
            arg_types_for_deduction = list(arg_types)
```